### PR TITLE
Add some tests to SPTPersistentCache

### PR DESCRIPTION
**Pull request by 8W9aG**
- Introduce the dispatch on main thread trick for asynchronous calls (like we did in data loader)

### CODE REVIEW

**Comment by 8W9aG** [comment]
@spotify/objc-dev

File: Sources/SPTPersistentCache.m [hunk 1]
```diff
@@ -910,14 +903,12 @@ - (void)dispatchEmptyResponseWithResult:(SPTPersistentCacheResponseCode)result
...
-    if (callback != nil) {
+    [self dispatchBlock:^ {
         SPTPersistentCacheResponse *response = [[SPTPersistentCacheResponse alloc] initWithResult:result
                                                                                             error:nil
                                                                                            record:nil];
-        dispatch_async(queue, ^{
-            callback(response);
-        });
-    }
+        callback(response);
```
Comment (rastersize) [inline Sources/SPTPersistentCache.m:910]: No `nil` check around this callback block? What if someone passes `nil`?

**Comment by rastersize** [comment]
:+1: when the [comment](https://github.com/spotify/SPTPersistentCache/pull/14#discussion_r53552184) has addressed.

File: Sources/SPTPersistentCache.m (hunk 1 above)
Comment (8W9aG) [inline Sources/SPTPersistentCache.m:910]: The static analyzer picks up when we attempt to call nil on a block without checking, and all our blocks have been fully qualified in this method. There isn't in the error dispatch method which is why we need the check there

**Pull request merged**